pkg/component/runtime/runtime_comm.go

```diff
@@ -137,6 +145,30 @@ func (c *runtimeComm) CheckinExpected(expected *proto.CheckinExpected) {
 	} else {
 		expected.AgentInfo = nil
 	}
+
+	// we need to determine if the communicator is currently in the initial observed message path
+	// in the case that it is we send the expected state over a different channel
+	c.initCheckinObservedMx.Lock()
+	initObserved := c.initCheckinObserved
+	expectedCh := c.initCheckinExpectedCh
+	if initObserved != nil {
+		// the next call to `CheckinExpected` must be from the initial `CheckinObserved` message
+		if observed != initObserved {
```

Comment:
Just to double check, we only care about equality of the references here? I know these are both pointers so I assume so.

If you need to compare the proto messages by value you need to use `proto.Equal(observed, initObserved)` otherwise this comparison will have unpredictable results. 

It _looks_ fine but we had a previous bug where we failed to use `proto.Equal` so I want to confirm the intent here.

Reply:
Yes we only care about comparing the reference. We want to be sure that the observed message is the exact message that was the initial message, not just the same contents.